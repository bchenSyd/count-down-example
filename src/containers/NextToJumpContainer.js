import React, { Component } from "react";
import fetch from "../common/fetch";
import Event from "../components/Event";
import { basePath } from "../config";


class NextToJumpContainer extends Component {
  state = {
    events: []
  };

  interval = undefined;
  componentDidMount() {
    const self = this;
    const fetchEvents = () =>
      fetch(`${basePath}/events`)
        .then(res => {
          const { success, result } = res;
          if (!success) {
            throw "fetch failed";
          }
          self.setState({
            events: [...result]
          });
        })
        .catch(err => {
          console.warn("needs to direct to error page", err);
        });

    fetchEvents();
    this.interval = setInterval(fetchEvents, 5000);
  }

  componentWillUnmount() {
    if (this.interval) {
      clearInterval(this.interval);
    }
  }

  render() {
    const { events } = this.state;
    return (
      <div className="next2Jump">
        <div>
          <label>Next to Jump</label>
        </div>
        <div className="events">
          {events.map(e => (
            <Event key={`event-${e.EventID}`} event={e} />
          ))}
        </div>
      </div>
    );
  }
}

export default NextToJumpContainer;
