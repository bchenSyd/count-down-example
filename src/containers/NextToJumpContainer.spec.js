import React from "react";
import renderer from "react-test-renderer";
import NextToJumpContainer from "./NextToJumpContainer";

jest.mock("../components/Event", () => "Event");
const response1 = {
  result: [
    {
      EventID: 30717084,
      EventName: "Race 3",
      EventTypeDesc: "Greyhounds",
      MasterEventName: "Races",
      AdvertisedStartTime: "2018-09-20T00:22:09.753Z",
      Venue: {
        Venue: "Daytona Beach",
        StateCode: null,
        Slug: "daytona-beach"
      }
    }
  ],
  success: true
};
const response2 = {
  result: [
    {
      EventID: 30717085,
      EventName: "Race 4",
      EventTypeDesc: "Greyhounds",
      MasterEventName: "Races",
      AdvertisedStartTime: "2018-09-20T00:32:09.753Z",
      Venue: {
        Venue: "Daytona Beach",
        StateCode: null,
        Slug: "daytona-beach"
      }
    }
  ],
  success: true
};

describe("next to jump", () => {
  it("should render correctly", () => {
    fetch = jest.fn().mockReturnValueOnce(Promise.resolve(response1));
    const component = renderer.create(<NextToJumpContainer />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
