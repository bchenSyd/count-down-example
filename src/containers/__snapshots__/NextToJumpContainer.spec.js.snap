// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`next to jump should render correctly 1`] = `
<div
  className="next2Jump"
>
  <div>
    <label>
      Next to Jump
    </label>
  </div>
  <div
    className="events"
  />
</div>
`;
